Vagrantfile

```diff
@@ -211,11 +217,20 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       if vmName == "node-%02d" % (i - 1)
         kHost.trigger.after [:up] do
           info "Waiting for Kubernetes minion [node-%02d" % (i - 1) + "] to become ready..."
-          system <<-EOT.prepend("\n\n") + "\n"
-            until curl -o /dev/null -sIf http://#{BASE_IP_ADDR}.#{i+100}:10250; do \
-              sleep 1;
-            done;
-          EOT
+          j, uri, hasResponse = 0, URI("http://#{BASE_IP_ADDR}.#{i+100}:10250"), false
+          loop do
+            j += 1
+            begin
+              res = Net::HTTP.get_response(uri)
+              hasResponse = true
+            rescue Net::HTTPBadResponse
+              hasResponse = true
```

Comment:
Here the Net::HTTPBadResponse is a valid response, because http://172.17.8.103:10250 does not give a valid http response. The exception means the socket connects and the node is up.